logic/context.py

```
@@ -0,0 +1,33 @@
+
+# Store embeddings globally
+note_embeddings = []
+
+def embed(text):
+    return model.encode(text)
+
+def cosine_similarity(a, b):
+    return np.dot(a, b) / (np.linalg.norm(a) * np.linalg.norm(b))
+
+def update_embeddings(notes):
+    global note_embeddings
```

Ruff (0.15.7)

[warning] Using the global statement to update `note_embeddings` is discouraged

(PLW0603)